ouroboros-consensus/docs/ReleaseProcess.md

```diff
@@ -0,0 +1,140 @@
+- RULE: The pending entries MUST NOT assume any specific version number of the previous release or the next release, because the entry may be backported etc.
+
+*Remark*.
+Backporting a PR wouldn't generate any conflicts in the changelog entries (by `scriv`'s design), so the author won't necessarily be prompted to update explicit version numbers.
+Hence we forbid them, via the review process.
+To explicitly clarify: it's fine to refer to historical explicit version numbers, but not in such a way that assumes anything about how many or which versions are between that explicit version and the changelog entry that mentions it.
+
+## Rules for Releases
+
+Infinitely often, the Consensus team will decide that FOO and/or BAR are ready for a next release (either from the main branch or from a release branch).
+IE We will eventually decide that the code of that package on the tip commit of a branch should be the code of our next release of that package from that branch.
+We prepare that release as follows.
+
+- For each package we are including in the release, we review its pending changelog entries.
+- RULE: We update the declared version (ie in its `.cabal` file) of each package we are including in the release based on the content of those pending changelog entries.
+    - (Note that there's at least one package, since otherwise we wouldn't be mkaing a release.)
+    - (Note that there must be some alterations to each package we are including in the release, since otherwise we wouldn't be including it in the release---but mind the _bundles_ mentioned below.)
+    - Let X.Y.Z be the version of the package currently declared on this branch.
+    - RULE: If any alteration was major-level, then we bump to (X+1).0.0.
+    - RULE: Else if any alteration was minor-level, then we bump to X.(Y+1).0.
+    - RULE: Otherwise all alterations were patch-level, so we bump to X.Y.(Z+1).
+- RULE: We merge one final PR, which MUST do exactly the following and nothing more.
+    - RULE: It updates the versions of the packages being released as described above.
+    - RULE: It flushes the pending changelog entries of each package being released into the `CHANGELOG.md` file for that package.
+- RULE: We tag that resulting merge commit as release-PKG-A.B.C; one tag per package PKG that is being released (ie FOO and/or BAR).
+- RULE: If C is 0, then we also create the release branch release-PKG-A.B.x from this new commit.
+- RULE: Finally, we announce this commit hash as the new release of these packages.
+  EG We insert these package's new versions into Hackage, [CHaP](https://github.com/input-output-hk/cardano-haskell-packages), etc.
+
+*Remark*.
+To explicitly clarify: after a release of a package, there will be zero pending changelog entries for that package; they were destroyed when incorprated into the `CHANGELOG.md` file making the release.
```

Comment:
```suggestion
To explicitly clarify: after a release of a package, there will be zero pending changelog entries for that package; they were merged into the `CHANGELOG.md` file when making the release.
```

Reply:
I expanded this sentence to be much more explicit.